### DISCUSSION TITLE
Using custom .env files.

### DISCUSSION THREAD

**Discussion by baquedano** [vercel/next.js#14609] [Help]
We were using dotenv to load or .env files, but it seems now it loads the .env files based off off the NODE_ENV variable, and in our case using the NODE_ENV="staging" is not going to work since we're using material-ui, which requires NODE_ENV to be "production" for certain functionality (this wouldn't be a problem is that one was configurable too).
So what we were doing  before was defining an APP_ENV="staging" and using it to load our .env files. is there any way to achieve this with the current version of nextjs?
if not, is there a way to turn off the automatic .env loading and allow Dotenv to do it instead?

**Comment by rafaelalmeidatk** [comment]
~~If you have `dotenv` in your `dependencies` or `devDependencies` list, Next will disable the automatic .env loading, so just by installing `dotenv` you should be good.~~ (no longer true)

To avoid Next from reading your `.env` files, you could change their name and use `dotenv` to load the file like this:

```js
require('dotenv').config({ path: '.custom_env_filename' })
```

But in general I recommend having only one .env file that has different contents depending on the environment it is at. Ref: https://12factor.net/config

---

And yeah, it is a bad idea to do `NODE_ENV=staging` 😉: https://rafaelalmeidatk.com/blog/why-you-should-not-use-a-custom-value-with-node-env

**Reply by timneutkens** [reply]
> If you have dotenv in your dependencies or devDependencies list, Next will disable the automatic .env loading, so just by installing dotenv you should be good.

This is no longer the case, dotenv is always loaded (as users ran into trouble if we didn't)

**Reply by rafaelalmeidatk** [reply]
Oh ops, I just learned I shouldn't use the files in the `error` folder as reference since they can be there for an old version 😅

**Reply by rafaelalmeidatk** [reply]
@baquedano I updated the answer 👍
